packages/odyssey-react/src/components/Tooltip/Tooltip.module.scss

```diff
@@ -0,0 +1,221 @@
+/*!
+ * Copyright (c) 2021-present, Okta, Inc. and/or its affiliates. All rights reserved.
+ * The Okta software accompanied by this notice is provided pursuant to the Apache License, Version 2.0 (the "License.")
+ *
+ * You may obtain a copy of the License at http://www.apache.org/licenses/LICENSE-2.0.
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *
+ * See the License for the specific language governing permissions and limitations under the License.
+ */
+
+.root {
+  @include border-radius($base-border-radius);
+
+  display: block;
+  visibility: hidden;
+  position: absolute;
+  z-index: 1;
+  padding-block: $spacing-xs;
+  padding-inline: $spacing-s;
+  transition: opacity $base-duration $base-timing 1s;
+  border: 1px solid cv('gray', '900');
+  opacity: 0;
+  background: cv('gray', '900');
+  color: cv('gray', '000');
+  font-size: $size-body-caption;
+  font-weight: 600;
+  line-height: $title-line-height;
+  white-space: nowrap;
+  pointer-events: none;
+
+  &::after,
+  &::before {
+    content: '';
+    position: absolute;
+    width: 0;
+    height: 0;
+    border: solid transparent;
+  }
+
+  &::after {
+    border-width: calc(#{$spacing-xs} - 2px);
+  }
+
+  &::before {
+    border-width: $spacing-xs;
+  }
+
+  *:focus + &,
```

Comment:
hmmm I'm not sure about this code. I like that it is a single line of CSS from a wow cool perspective, but I don't love that we're using a universal selector against the entire DOM and relying on the structure and relation of nodes to control this style application. Tooltip shouldn't have to care about whether any adjacent sibling has focus, it feels like the type of contextual knowledge we don't want our components to be concerned with or affected by.

Reply:
I agree we should move away from this selector, but our current Tooltip component will break without it. Tooltip needs to know when any adjacent, focusable sibling has `:focus` in order to trigger the tip. This behavior doesn't need to be global, as any siblings will also be a child of `.hasTooltip`.

For this scope of work, I could rewrite the old selector to only match cases where both children have `.hasTooltip` as a sibling. Something like `.hasTooltip *:focus + .root` should work.

Another solution would be passing a class to `{clone}` and selecting for that instead, which would be much more explicit.

Reply:
Let's scope the universal selector as a indirect descendant of the tooltip class for now. That will provide the most robust solution as it still works for cloned children that have deeply nested focusable content. Looking forward we'll want to revisit Tooltip and build out a controlled implementation where the tip can be opened imperatively and we can revisit this CSS driven approach.